fms/modules/attention.py

```diff
@@ -292,11 +292,12 @@ def __init__(
             position_encoder,
             gain,
         )
+        self.pretp_kvheads = kvheads
```

Comment:
nit: for some reason this took me a minute... not sure if maybe `pre_tp_kvheads` would be clearer